geowebcache/core/src/main/java/org/geowebcache/storage/blobstore/file/FilePathGenerator.java

```diff
@@ -107,42 +106,29 @@ public File tilePath(TileObject tile, MimeType mimeType) {
         return tileFile;
     }
 
-    protected static String buildKey(String parametersKvp) {
+    public static String buildKey(String parametersKvp) {
         return DigestUtils.shaHex(parametersKvp);
     }
     
     /**
      * Returns the parameters identifier for the given parameters map
      * @param parameters
      * @return
+     * @deprecated Use {@link ParametersUtils#getParametersId(Map<String, String>)} instead
      */
     public static String getParametersId(Map<String, String> parameters) {
-        if(parameters == null || parameters.size() == 0) {
-            return null;
-        }
-        String parametersKvp = getParametersKvp(parameters);
-        return buildKey(parametersKvp);
+        return ParametersUtils.getId(parameters);
```

Comment:
This calls the utils which in turn calls back this class... seems a bit weird to me... wondering if it was not best to avoid making buildKey public and just call the utils there too (like, having utils expose a buildKey method and delegate). Maybe call DigestUtils.sha1Hex there.